beater/server.go

```diff
@@ -235,6 +235,7 @@ func (s server) run() error {
 	if s.jaegerServer != nil {
 		g.Go(s.jaegerServer.Serve)
 	}
+	runningc <- struct{}{}
```

Comment:
We don't really know that the listener has been created yet. It's possible that `s.httpServer.start` on line 231 hasn't been scheduled yet, or that the listener it creates hasn't been created, by the time we signal `runningc`.

How about we move the listener creation out of `httpServer.start` and into the `newHTTPServer` function? That way when `newBaseRunServer` returns we know the listener has been created.